src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs

```diff
@@ -557,19 +558,51 @@ public static ExecCredentialResponse ExecuteExternalCommand(ExternalExecution co
                 throw new KubeConfigException($"external exec failed due to: {ex.Message}");
             }
 
-            var stdout = process.StandardOutput.ReadToEnd();
-            var stderr = process.StandardError.ReadToEnd();
-            if (string.IsNullOrWhiteSpace(stderr) == false)
+            var stderrBuilder = new StringBuilder();
+
+            var stdOutTask = process.StandardOutput.ReadToEndAsync(); // Assumes process exits
+
+            var buffer = new char[1024];
+
+            var stopwatch = Stopwatch.StartNew();
+
+            // Wait for a maximum of 5 seconds, if a response takes longer probably something went wrong...
+            while (!stdOutTask.IsCompleted && stopwatch.Elapsed < TimeSpan.FromSeconds(5))
```

Comment:
you can do the pattern like below without have to loop and sleep, which will always cause thread to sleep 500ms

```
t = task.create { readstdout and stderr }
task.waitany(delay(timeout), t)
```

Reply:
I tried that logic last night, however, it introduced another issue and it still needed a loop. Let me bring it back and see if I might have missed something as it was rater late.

Reply:
add ct to readasync